docs/sdk/wrapped-keys/generating-wrapped-key.md

```diff
@@ -0,0 +1,315 @@
+- [Session Signatures](../../sdk/authentication/session-sigs/intro)
+
+## `generatePrivateKey`'s Interface
+
+<!-- TODO Update once Wrapped Keys PR is merged -->
+[Source code](https://github.com/LIT-Protocol/js-sdk/blob/b80cee7035639ef4739c6190812eecbf2d2dab2e/packages/wrapped-keys/src/lib/wrapped-keys.ts#L29-L86)
+
+```ts
+/**
+ *
+ * Generates a random Solana/EVM private key inside the corresponding Lit Action and returns the publicKey of the random private key. We don't return the generated wallet address since it can be derived from the publicKey
+ *
+ * @param pkpSessionSigs - The PKP sessionSigs used to associated the PKP with the generated private key
+ * @param network - The network for which the private key needs to be generated. This is used to call different Lit Actions since the keys will be of different types
+ * @param litNodeClient - The Lit Node Client used for executing the Lit Action
+ *
+ * @returns { Promise<GeneratePrivateKeyResponse> } - The publicKey of the generated random private key along with the PKP EthAddres associated with the Wrapped Key
+ */
+export async function generatePrivateKey({
+  pkpSessionSigs,
+  network,
+  litNodeClient,
+}: GeneratePrivateKeyParams): Promise<GeneratePrivateKeyResponse>
+```
+
+### Parameters
+
+#### `pkpSessionSigs`
+
+When a Wrapped Key is generated, it's encrypted with the following [Access Control Conditions](../../sdk/access-control/evm/basic-examples):
+
+```ts
+[
+    {
+        contractAddress: '',
+        standardContractType: '',
+        chain: CHAIN_ETHEREUM,
+        method: '',
+        parameters: [':userAddress'],
+        returnValueTest: {
+        comparator: '=',
+        value: pkpAddress,
+        },
+    },
+];
+```
+
+where `pkpAddress` is the addressed derived from the `pkpSessionSigs`. This restricts the decryption of the Wrapped Key to only those whom can generate valid Authentication Signatures from the PKP which generated the Wrapped Key.
+
+A valid `pkpSessionSigs` object can be obtained using the [getPkpSessionSigs](https://v6-api-doc-lit-js-sdk.vercel.app/classes/lit_node_client_src.LitNodeClientNodeJs.html#getPkpSessionSigs) helper method available on an instance of [LitNodeClient](https://v6-api-doc-lit-js-sdk.vercel.app/classes/lit_node_client_src.LitNodeClient.html). We dive deeper into obtaining a `pkpSessionSigs` using `getPkpSessionSigs` in the [Generating PKP Session Signatures](#generating-pkp-session-signatures) section of this guide.
+
+#### `network`
+
+This parameter dictates what elliptic curve is used to generate the private key. It must be one of the supported Wrapped Keys [Networks](https://github.com/LIT-Protocol/js-sdk/blob/b80cee7035639ef4739c6190812eecbf2d2dab2e/packages/wrapped-keys/src/lib/constants.ts#L5) which currently consists of:
+
+  - `evm` This will generate a private key using the ECDSA curve.
+  - `solana` This will generate a private key using the Ed25519 curve.
```

Comment:
wondering if there is a better way to frame this to encompass all non-evm chains that use eddsa (like Sui for example)